### PR TITLE
BenchmarkDriver Strangler replaces Benchmark_Driver run

### PR DESCRIPTION
**Pull request by palimondo**
This is a followup to PR #18719 that concludes the [StranglerApplication](https://www.martinfowler.com/bliki/StranglerApplication.html) of the `Benchmark_Driver`'s **`run`** command. The remaining free-standing functions implementing the `run` command were replaced with methods on `BenchmarkDriver` class with full unit test coverage.

The functionality remains unchanged with the exception of cleaning up the last "Totals" line of the log reports, after removing the bogus aggregate statistics. It now simply states the number of executed benchmarks:
````
Total performance tests executed: 1
````

I've structured the PR as a series of small, incremental changes with detailed descriptions in the commit messages, it is therefore best [reviewed sequentially by individual commits](https://github.com/apple/swift/pull/18924/commits/0d64386b530b128564f266dbfcb88f8b6cbc2150).

### CODE REVIEW

**Comment by palimondo** [comment]
@eeckstein Please review 🙏

**Comment by lorentey** [comment]
@swift-ci please benchmark

**Comment by lorentey** [comment]
@swift-ci please smoke test

File: benchmark/scripts/compare_perf_tests.py [hunk 1]
```diff
@@ -308,7 +308,7 @@ def _reset(self):
 
     # Parse lines like this
     # #,TEST,SAMPLES,MIN(μs),MAX(μs),MEAN(μs),SD(μs),MEDIAN(μs)
-    results_re = re.compile(r'(\d+[, \t]*\w+[, \t]*' +
+    results_re = re.compile(r'([ ]*\d+[, \t]*\w+[, \t]*' +
```
Comment (eeckstein) [inline benchmark/scripts/compare_perf_tests.py]: nitpick: brackets are not required around the single space

File: benchmark/scripts/Benchmark_Driver [hunk 2]
```diff
@@ -55,6 +55,9 @@ class BenchmarkDriver(object):
         self.tests = tests or self._get_tests()
         self.parser = parser or LogParser()
         self.results = {}
+        # Set a constant hash seed. Some tests are currently sensitive to
+        # fluctuations in the number of hash collisions.
+        os.environ['SWIFT_DETERMINISTIC_HASHING'] = '1'
```
Comment (eeckstein) [inline benchmark/scripts/Benchmark_Driver]: That's an important change. It should be in a separate commit or at least be mentioned in the commit message

**Review by eeckstein — APPROVED** [review]
lgtm, thanks!
Just a few minor comments.

File: benchmark/scripts/Benchmark_Driver (hunk 2 above)
Comment (palimondo) [reply benchmark/scripts/Benchmark_Driver]: I’m not sure what you mean. Deterministic hashing env variable was [added some 5 months ago by @lorentey](https://github.com/apple/swift/blame/master/benchmark/scripts/Benchmark_Driver#L420). I’ve just added a unit test for it and moved the implementation into `BenchmarkDriver` class.

File: benchmark/scripts/compare_perf_tests.py (hunk 1 above)
Comment (palimondo) [reply benchmark/scripts/compare_perf_tests.py]: I did not know that space and a star would work. This way it is in same style as all the other repeated groups. Do you want me to change it?

**Comment by palimondo** [comment]
@eeckstein Benchmarks finished successfully. Can you merge this as is or should I change something?
